Multisig/Features/Security/ItemSearchQuery.swift

```diff
@@ -0,0 +1,153 @@
+                kSecReturnRef: true,
+            ]
+        case .ecKeyPair:
+            result = [:]
+        }
+
+        return result
+    }
+
+    func createAttributesForItem() throws -> NSDictionary {
+        var result: NSMutableDictionary = [:]
+        switch self {
+        case let .generic(id, service, data):
+            result = [
+                kSecAttrService: service,
+                kSecAttrAccount: id,
+                kSecClass: kSecClassGenericPassword,
+                kSecReturnAttributes: false,
+                kSecReturnData: true,
+            ]
+            if let data = data {
+                result[kSecValueData] = data
+            }
+
+        case let .enclaveKey(tag, password, access):
+            let accessControl = try accessControl(flags: .privateKeyUsage.union(access!))
+
+            let privateKeyAttrs: NSMutableDictionary = [
+                kSecAttrIsPermanent: true,
+                kSecAttrApplicationTag: tag.data(using: .utf8)!,
+                kSecAttrAccessControl: accessControl
+            ]
+            if let context = LAContext(password: password) {
+                privateKeyAttrs[kSecUseAuthenticationContext] = context
+            }
+            result = [
+                kSecAttrKeyType: kSecAttrKeyTypeECSECPrimeRandom,
+                kSecAttrKeySizeInBits: 256,
+                kSecAttrTokenID: kSecAttrTokenIDSecureEnclave,
+                kSecPrivateKeyAttrs: privateKeyAttrs
+            ]
+        case .ecKeyPair:
+            result = [
+                kSecAttrKeyType: kSecAttrKeyTypeECSECPrimeRandom,
+                kSecAttrKeySizeInBits: 256,
+                kSecAttrKeyClass: kSecAttrKeyClassPrivate
+            ]
+        case let .ecPubKey(tag, pubKey):
+            result = [
+                kSecClass: kSecClassKey,
+                kSecAttrKeyClass: kSecAttrKeyClassPublic,
+                kSecAttrApplicationTag: tag.data(using: .utf8)!,
+                kSecAttrKeyType: kSecAttrKeyTypeECSECPrimeRandom,
+                kSecReturnRef: true
+            ]
+            if let pubKey = pubKey {
+                result[kSecValueRef] = pubKey
+            }
+        }
+        return result
+    }
+
+    // create access control flags with params
+    fileprivate func accessControl(flags: SecAccessControlCreateFlags) throws -> SecAccessControl {
+        // SWIFT: can't extend SecAccessControl (compiler error that extensions of CF classes are not supported).
+
+        var accessError: Unmanaged<CFError>?
+        guard let access = SecAccessControlCreateWithFlags(
+                kCFAllocatorDefault,
+                kSecAttrAccessibleAfterFirstUnlock,
+                flags,
+                &accessError
+        )
+        else {
+            throw accessError!.takeRetainedValue() as Error
+        }
+        return access
+    }
+
+}
+
+func ==(left: ItemSearchQuery, right: ItemSearchQuery) -> Bool {
```

Comment:
the body of this function doesn't make sense to me - why is any ecKeyPair is the same?

Reply:
I replaced its invocation with a simple switch statement wit a default branch.